internal/core/items/text.rs

```diff
@@ -612,7 +616,7 @@ impl TextInput {
 
         if reset_preferred_x_pos {
             self.preferred_x_pos
-                .set(window.text_input_position_for_byte_offset(self, new_cursor_pos).x);
+                .set(window.text_input_cursor_rect_for_byte_offset(self, new_cursor_pos).origin.x);
```

Comment:
I think this should be done in set_cursor_position. It can take ab argument If it should be updated or not

Reply:
that was easy to do, so i added it to this pr.